src/content/docs/code-push/rollback.mdx

```diff
@@ -47,3 +47,36 @@ If you app needs to download an older patch as part of a rollback,
 installing that patch will count against your monthly patch installs. If you
 have no remaining patch installs, the app will revert to the base release.
 :::
+
+## Patch Integrity and Automatic Rollback
+
+Shorebird includes safety features to prevent needing to issue the manual roll
+back of a patch.
+
+`shorebird patch` also generates a hash of your patch file which is stored
+separately from the patch and checked every time your patch is installed, or
+booted from on the user's device.
+
+Customers are also encouraged to use Shorebird's [patch-signing
```

Comment:
Not sure if necessary, but I think we still need to support this for hybrid app targets  https://github.com/shorebirdtech/shorebird/issues/2162

Reply:
That's fine, we'll add it when requested.

Reply:
My suggestion was more "should we mention it here so someone who reads this and tries isn't confused"

Reply:
Nah, I think that goes in the signing docs themselves.  We also just generally need a "make our system awesome for hybrid apps" pass over our whole product, that I think comes later as we're more established.